samples/azure/azure-service-fabric-routing/Core_6/Shared/ReceiverSideDistribution/Pipeline/Forwarder.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using NServiceBus;
+
+class Forwarder
+{
+    LogicalAddress logicalAddress;
+    HashSet<string> knownPartitionKeys;
+    Func<LogicalAddress, string> addressTranslator;
+
+    public Forwarder(HashSet<string> knownPartitionKeys, Func<LogicalAddress, string> addressTranslator, LogicalAddress logicalAddress)
+    {
+        this.knownPartitionKeys = knownPartitionKeys;
+        this.addressTranslator = addressTranslator;
+        this.logicalAddress = logicalAddress;
+    }
+
+    public Task Forward(IMessageProcessingContext context, string messagePartitionKey)
+    {
+        if (!knownPartitionKeys.Contains(messagePartitionKey))
```

Comment:
@ramonsmits there's no read. We only check if the value is in the partition list we know about and that's it.